build/helper/documentation_helper.py

```diff
@@ -505,14 +533,26 @@ def as_rest_table(data, header=True):
 
 def _remove_trailing_whitespace(s):
     '''Removes trailing whitespace and empty lines in multi-line strings. https://stackoverflow.com/a/17350806/316875'''
-    return re.sub(r'\s+$', '', s, flags=re.M)
+    initial_lines = s.strip().splitlines()
+    fixed_lines = []
+    blank_lines = 0
+    for l in initial_lines:
+        stripped_line = l.strip()
+        if len(stripped_line) == 0 and blank_lines == 0:
+            fixed_lines.append(stripped_line)
+            blank_lines = 1
+        if len(stripped_line) > 0:
+            fixed_lines.append(stripped_line)
+            blank_lines = 0
+
+    return fixed_lines
 
 
 def assert_rst_strings_are_equal(expected, actual):
     '''Asserts rst formatted strings (multiline) are equal. Ignores trailing whitespace and empty lines.'''
-    expected = _remove_trailing_whitespace(expected).splitlines()
-    actual = _remove_trailing_whitespace(actual).splitlines()
-    assert len(expected) == len(actual), 'Strings have different number of non-empty lines.'
+    expected = _remove_trailing_whitespace(expected)
+    actual = _remove_trailing_whitespace(actual)
+    # assert len(expected) == len(actual), 'Strings have different number of non-empty lines.'
```

Comment:
remove

Reply:
While asserting that actual and expected strings have the same number of lines is more correct, it actually makes it harder to debug differences.

Removed commented out assert